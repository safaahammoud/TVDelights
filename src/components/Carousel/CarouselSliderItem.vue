<template>
    <div class="carousel__item">
        <div class="carousel__item__image-wrapper">
            <img
                :src="props.image || defaultImagePlaceHolder"
                :alt="props.name"
                class="carousel__item__image-wrapper__image"
            />
        </div>

        <div class="carousel__item__content-wrapper">
            <h3
                class="carousel__item__title"
                v-text="props.name"
            />

            <div
                v-if="props.rating"
                class="carousel__item__rating"
            >
                <StarRating :rate="props.rating"/>
            </div>

            <div
                v-if="props.genres.length"
                class="carousel__item__genres"
            >
                <PrimeTag
                    v-for="genre in props.genres"
                    :key="genre"
                    :value="genre"
                    rounded
                    severity="info"
                    class="carousel__item__genres__tag"
                />    
            </div>
        </div>
    </div>
</template>

<script setup lang="ts">
import defaultImagePlaceHolder from '@/assets/no-img-portrait-text.svg';
import StarRating from '@/components/StarRating.vue';

const props = defineProps<{
    name: string;
    image: string;
    rating: number;
    genres: string[];
}>();
</script>

<style lang="scss" scoped>
.carousel {
    &__item__content-wrapper {
        padding: 1rem;
    }

    &__item {
        height: 100%;
        display: flex;
        flex-direction: column;
        justify-content: space-between;
        margin: 0 0.5rem !important;
        text-align: center !important;
        border-radius: var(--border-radius) !important;
        border: 1px solid var(--surface-border) !important;

        &__image-wrapper {
            margin-bottom: 1rem;
    
            &__image {
                width: 100%;
            }
        }

        &__title {
            margin: 0 0 1rem 0;
        }
        
        &__title:hover {
            text-decoration: underline;
        }

        &__genres {
            &__tag {
                margin: 0 0.5rem 0.5rem 0;
            }
        }

        &__rating {
            display: flex;
            justify-content: center;
            margin-bottom: 1rem;
        }
    }
}
</style>