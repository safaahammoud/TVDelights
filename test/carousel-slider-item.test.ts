import { mount } from '@vue/test-utils';
import { describe, expect, test } from 'vitest';

import CarouselSliderItem from '../src/components/Carousel/CarouselSliderItem.vue';

describe('CarouselSliderItem', () => {
  test('mount component and renders properly', async () => {
    const mockItem = {
      name: 'test',
      image: 'test-image-url',
      genres: ['comedy'],
      rating: 6,
    };
    const wrapper = mount(CarouselSliderItem, {
      props: mockItem,
    });
    
    expect(wrapper.find('h3').text()).toContain(mockItem.name);
    expect(wrapper.find('.carousel__item__image-wrapper__image').attributes('src')).toBe('test-image-url');
    expect(wrapper.find('.carousel__item__rating').text()).toContain('6');
  });
});
