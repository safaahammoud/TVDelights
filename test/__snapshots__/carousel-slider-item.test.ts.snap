// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CarouselSliderItem > mount component 1`] = `
"<div data-v-338fd4e5="" class="carousel__item">
  <div data-v-338fd4e5="" class="carousel__item__image-wrapper"><img data-v-338fd4e5="" src="/src/assets/no-img-portrait-text.svg" alt="tv show image" class="carousel__item__image-wrapper__image"></div>
  <div data-v-338fd4e5="" class="carousel__item__content-wrapper">
    <h3 data-v-338fd4e5="" class="carousel__item__title">test</h3>
    <div data-v-338fd4e5="" class="carousel__item__rating">
      <div data-v-11c024fb="" data-v-338fd4e5="" class="rating-wrapper"><i data-v-11c024fb="" class="rating-wrapper__star pi pi-star-fill"></i>
        <h4 data-v-11c024fb="" class="rating-wrapper__rating">6</h4>
      </div>
    </div>
    <div data-v-338fd4e5="" class="carousel__item__genres">
      <primetag data-v-338fd4e5="" rounded="" severity="info" class="carousel__item__genres__tag" value="comedy"></primetag>
    </div>
  </div>
</div>"
`;

exports[`CarouselSliderItem > mount component and renders properly 1`] = `
"<div data-v-338fd4e5="" class="carousel__item">
  <div data-v-338fd4e5="" class="carousel__item__image-wrapper"><img data-v-338fd4e5="" src="[object Object]" alt="tv show image" class="carousel__item__image-wrapper__image"></div>
  <div data-v-338fd4e5="" class="carousel__item__content-wrapper">
    <h3 data-v-338fd4e5="" class="carousel__item__title">test</h3>
    <div data-v-338fd4e5="" class="carousel__item__rating">
      <div data-v-11c024fb="" data-v-338fd4e5="" class="rating-wrapper"><i data-v-11c024fb="" class="rating-wrapper__star pi pi-star-fill"></i>
        <h4 data-v-11c024fb="" class="rating-wrapper__rating">6</h4>
      </div>
    </div>
    <div data-v-338fd4e5="" class="carousel__item__genres">
      <primetag data-v-338fd4e5="" rounded="" severity="info" class="carousel__item__genres__tag" value="comedy"></primetag>
    </div>
  </div>
</div>"
`;

exports[`mount component 1`] = `
"<div data-v-338fd4e5="" class="carousel__item">
  <div data-v-338fd4e5="" class="carousel__item__image-wrapper"><img data-v-338fd4e5="" src="/src/assets/no-img-portrait-text.svg" alt="tv show image" class="carousel__item__image-wrapper__image"></div>
  <div data-v-338fd4e5="" class="carousel__item__content-wrapper">
    <h3 data-v-338fd4e5="" class="carousel__item__title"></h3>
    <div data-v-338fd4e5="" class="carousel__item__rating">
      <div data-v-11c024fb="" data-v-338fd4e5="" class="rating-wrapper"><i data-v-11c024fb="" class="rating-wrapper__star pi pi-star-fill"></i>
        <h4 data-v-11c024fb="" class="rating-wrapper__rating">8</h4>
      </div>
    </div>
    <!--v-if-->
  </div>
</div>"
`;
